hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/common/AbstractStreamWriteFunction.java

```diff
@@ -195,15 +194,13 @@ public void setOperatorEventGateway(OperatorEventGateway operatorEventGateway) {
   private void restoreWriteMetadata() throws Exception {
     boolean eventSent = false;
     for (WriteMetadataEvent event : this.writeMetadataState.get()) {
-      if (Objects.equals(this.currentInstant, event.getInstantTime())) {
-        // Reset taskID for event
-        event.setTaskID(taskID);
-        // The checkpoint succeed but the meta does not commit,
-        // re-commit the inflight instant
-        this.eventGateway.sendEventToCoordinator(event);
-        LOG.info("Send uncommitted write metadata event to coordinator, task[{}].", taskID);
-        eventSent = true;
-      }
+      // Reset taskID for event
```

Comment:
It's better to add the test cases for recommit to verify whether the recommit behavior could work well.

Reply:
`TestStreamWriteOperatorCoordinator#testRecommitWithPartialUncommittedEvents` should already test this logic.

Reply:
@danny0405, without this change, `testRecommitWithPartialUncommittedEvents` could also work well, therefore `testRecommitWithPartialUncommittedEvents` does not verify this change.